src/styles/progress-indicator.scss

```diff
@@ -2,93 +2,232 @@
+  --containerBorderColor: var(--fdProgress_Indicator_Container_Border_Neutral, var(--sapField_BorderColor));
+  --iconDisplay: none;
+  --iconContent: '';
+  --labelMarginStart: var(--fdProgress_Indicator_Label_Margin_Start);
+  --labelMarginEnd: var(--fdProgress_Indicator_Label_Margin_End);
+
+  @include fd-reset();
+
   width: 100%;
-  margin: $fd-progress-indicator-margin;
+  height: var(--fdProgress_Indicator_Height);
+  margin: var(--rootMargin);
+  padding: var(--fdProgress_Indicator_Container_Margin);
 
   &.#{$fd-namespace}-popover {
-    margin: $fd-progress-indicator-margin;
+    margin: var(--rootMargin);
+  }
+
+  &--mobile {
+    --rootMargin: 0.875rem 0;
+  }
+
+  &--display {
+    margin: 0;
+
+    &.#{$fd-namespace}-popover {
+      margin: 0;
+    }
   }
 
   @each $set-name, $set-state in $fd-progress-indicator-states {
     &--#{$set-name} {
-      .#{$block}__progress-bar {
-        background-color: map_get($set-state, "background");
-      }
-
-      .#{$block}__label {
-        &::before {
-          vertical-align: middle;
-          font-family: "SAP-icons";
-          content: map_get($set-state, "icon");
-          margin-left: var(--fdProgress_Icon_Offset_Left);
-          margin-right: var(--fdProgress_Icon_Offset_Right);
-        }
-      }
+      --progressBarBackground: #{map_get($set-state, "background")};
+      --progressBarBorderColor: #{map_get($set-state, "borderColor")};
+      --containerBackground: #{map_get($set-state, "containerBackground")};
+      --containerBorderColor: #{map_get($set-state, "containerBorderColor")};
+      --iconContent: #{map_get($set-state, "icon")};
+      --iconDisplay: inline;
+      --iconColor: #{map_get($set-state, "iconColor")};
+      --labelMarginStart: 0;
     }
   }
 
-  .#{$block}__container {
+  &__container {
     @include fd-reset();
     @include fd-flex();
 
+    position: var(--fdProgress_Indicator_Container_Position, static);
     min-width: 4rem;
-    overflow: hidden;
+    width: var(--fdProgress_Indicator_Container_Width);
     border-radius: 0.5rem;
-    background: var(--sapField_Background);
-    height: $fd-progress-indicator-container-height;
-    border: 0.0625rem solid var(--sapField_BorderColor);
-    min-height: $fd-progress-indicator-container-height;
+    background: var(--containerBackground);
+    overflow: var(--fdProgress_Indicator_Container_Overflow);
+    height: var(--fdProgress_Indicator_Container_Height);
+    min-height: var(--fdProgress_Indicator_Container_Height);
+    border: 0.0625rem solid var(--containerBorderColor);
+
+    &::before,
+    &::after {
+      position: absolute;
+      content: '';
+      display: var(--fdProgress_Indicator_Progress_Bar_Boundary_Display, none);
+      background-color: var(--progressBarBackground);
+      width: 0.25rem;
+      height: 0.25rem;
+      border-radius: 50%;
+      top: 0.0005rem;
+    }
+
+    &::before {
+      left: 0.0005rem;
+    }
+
+    &::after {
+      right: 0.0005rem;
```

Comment:
The specs call for 0.063rem here, giving 1px between the dots and the edge of the indicator

Reply:
Yes, you are right, that should be 0, because border makes that 1px spacing